docs/src/developer_documentation/custom_tangent_type.md

```diff
@@ -737,7 +739,14 @@ end
 ```
 
 Now we can run it again and successfully check if all the tangent / fdata / rdata and other required functionality works correctly for the self-referential type A.
+We run the check for both a non-cyclic case to check our method implementations,
+as well as a cyclic case to make sure that our interactions with the caches are correct.
 
 ```@example custom_tangent_type
+# Non-cyclic A
 Mooncake.TestUtils.test_data(Random.default_rng(), A(1.0, A(2.0, A(3.0))))
+# Cyclic A
+cyclic_a = A(1.0, A(2.0))
+cyclic_a.a.a = cyclic_a
+Mooncake.TestUtils.test_data(Random.default_rng(), cyclic_a)
```

Comment:
Do you think whether it's better to split this into a new code block? Currently only the result for line 751 will show.